gunicorn/http/wsgi.py

```diff
@@ -360,12 +360,6 @@ def sendfile(self, respiter):
             offset = os.lseek(fileno, 0, os.SEEK_CUR)
             if self.response_length is None:
                 filesize = os.fstat(fileno).st_size
-
-                # The file may be special and sendfile will fail.
-                # It may also be zero-length, but that is okay.
-                if filesize == 0:
```

Comment:
Why don't we want this anymore? Is it because the implementations in the stdlib, gevent, and this PR all fall back to reading and sending chunks, rather than calling the underlying OS implementation?

Reply:
Right. In the case that `_sendfile_use_sendfile` gets used (the version of `socket.sendfile` that uses `os.sendfile`) the standard library and gevent implementations handle the 0 size case automatically. That's part `socket.sendfile` being a higher-level API than `os.sendfile.` The implementation provided here for eventlet, of course, doesn't use `os.sendfile`.